devicetypes/smartthings/ezex-smart-electric-switch.src/ezex-smart-electric-switch.groovy

```diff
@@ -0,0 +1,130 @@
+/**
+ *  Copyright 2019 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+    definition (name: "eZEX smart electric switch", namespace: "smartthings", author: "SmartThings", mnmn: "SmartThings", vid: "generic-switch-power-energy") {
+        capability "Energy Meter"
+        capability "Power Meter"
+        capability "Switch"
+        capability "Refresh"
+        capability "Polling"
+        capability "Sensor"
+
+        fingerprint profileId: "0104", deviceId:"0053", inClusters: "0000, 0003, 0004, 0006, 0B04, 0702", outClusters: "0019", manufacturer: "", model: "E240-KR116Z-HA", deviceJoinName: "eZEX smart electric switch"
+    }
+
+    tiles(scale: 2){
+        multiAttributeTile(name:"switch", type: "generic", width: 6, height: 4, canChangeIcon: true){
+                tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+                        attributeState("on", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00a0dc")
+                        attributeState("off", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff")
+                }
+        }
+        valueTile("power", "device.power", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} W'
+        }
+        valueTile("energy", "device.energy", decoration: "flat", width: 2, height: 2) {
+                state "default", label:'${currentValue} kWh'
+        }
+        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        standardTile("reset", "device.energy", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+                state "default", label:'reset kWh', action:"reset"
+        }
+
+        main(["switch"])
+        details(["switch","power","energy","refresh","reset"])
+    }
+}
+
+def parse(String description) {
+    log.debug "description is $description"
+    def event = zigbee.getEvent(description)
+    if (event) {
+        log.info event
+        sendEvent(event)
+    } else {
+        List result = []
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+         
+        if (descMap.cluster == 0x0702 && descMap.attrId == 0x0000) {
```

Comment:
This should still be moved into the loop below, and you need to use `clusterInt` and `attrInt` instead for the integer comparisons.